.github/bin/deprecations.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+log=$1
+
+safe_grep() {
+    grep "$@" || true
+}
+
+stats=$(safe_grep ' in io\.trino\.spi\..* has been deprecated$' "$log" | \
+    safe_grep -o "$PWD/[^ ]\+.java" | \
+    sed "s@$PWD/@@" | \
+    sed -E "s@/src/(main|test)/java/.*@:\1@" | sort | uniq -c | sort -nr)
+
+if [ -n "$stats" ]; then
+    grep ' in io\.trino\.spi\..* has been deprecated$' "$log" | sed 's/^\[WARNING\]/[ERROR]/'
```

Comment:
Do you know of something configurable enough to fit our needs? I played with error-prone a bit and it seems it's possible to build our own checker based on source & call-site of the API, plus maybe some other context from AST. But again it could be hard to maintain too.
https://github.com/nevillelyh/error-prone/commit/60ecb2709e60a18a9fb420f2b91268123e553da8